### PR TITLE
PENG-3689 - Introducing support for datasets functionality

### PR DESCRIPTION
**Pull request by cpontes-ns1**

### CODE REVIEW

File: ns1/rest/transport/requests.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def _rateLimitHeaders(self, headers):
             "remaining": int(headers.get("X-RateLimit-Remaining", 100)),
```
Comment (cpontes-ns1) [inline ns1/rest/transport/requests.py:45]: In my case, I have a response that is not JSON. Currently all responses are assumed to be JSON so I needed a way to handle this

File: ns1/dataset.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        """
+        return self.load(reload=True, callback=callback, errback=errback)
+
+    def load(self, id=None, callback=None, errback=None, reload=False):
+        """
+        Load dataset data from the API.
+        :param str id: dataset id to load
+        """
+        if not reload and self.data:
+            raise DatasetException("dataset already loaded")
```
[the remark this answers is not in the document]
Comment (cpontes-ns1) [reply ns1/dataset.py]: Fixed

File: ns1/dataset.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                return callback(ret)
+            else:
+                return ret
+
+        return Datasets(self.config).list(callback=success, errback=errback)
+
+    def retrieveReport(self, rp_id, dt_id=None, callback=None, errback=None):
+        """
+        Retrieves a generated report given a dataset id and a report id
+        :return: generated report
```
[the remark this answers is not in the document]
Comment (cpontes-ns1) [reply ns1/dataset.py:162]: Fixed

File: ns1/dataset.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        """
+        Create a new dataset. Pass a list of keywords and their values to
+        configure. For the list of keywords available for dataset configuration,
+        see :attr:`ns1.rest.datasets.Datasets.PASSTHRU_FIELDS`
+        :param str name: the name of the dataset
+        :param str datatype: datatype settings to define the type of data to be pulled
+        :param str repeat: repeat settings to define recurrent reports
+        :param str timeframe: timeframe settings for the data to be pulled
+        :param str export_type: output format of the report
+        :param str recipient_emails: list of user emails that will receive a copy of the report
```
[the remark this answers is not in the document]
Comment (cpontes-ns1) [reply ns1/dataset.py:117]: fixed

**Review by hmodi-ns1 — APPROVED** [review]

**Pull request merged**